kasa/smart/smartdevice.py

```diff
@@ -321,13 +321,11 @@ def get_module(self, module_type: type[ModuleT]) -> ModuleT | None: ...
     @overload
     def get_module(self, module_type: str) -> SmartModule | None: ...
 
-    def get_module(
-        self, module_type: type[ModuleT] | str
-    ) -> ModuleT | SmartModule | None:
+    def get_module(self, module_type: type[ModuleT] | str) -> ModuleT | Module | None:
         """Return the module from the device modules or None if not present."""
         if isinstance(module_type, str):
             module_name = module_type
-        elif issubclass(module_type, SmartModule):
+        elif issubclass(module_type, Module):
```

Comment:
Same as above for IotModules, should this remain as it was?

Reply:
As above this was needed because the common module which was passed in as a type (to be converted to string of __name__) was the base class type.  This is no longer an issue with the new typed Mapping and get_module has been removed.